src/util/logging.jl

```diff
@@ -1,51 +0,0 @@
-# Logging functionality
-
-# I/O without libuv, for use after STDOUT is finalized
-raw_print(msg::AbstractString...) =
-    ccall(:write, Cssize_t, (Cint, Cstring, Csize_t), 1, join(msg), length(join(msg)))
-raw_println(msg::AbstractString...) = raw_print(msg..., "\n")
-
-# safe version of `Base.print_with_color`, switching to raw I/O before finalizers are run
-# (see `atexit` in `__init_logging__`)
-const after_exit = Ref{Core.Core.Bool}(false)
-function safe_print_with_color(color::Union{Int, Symbol}, io::IO, msg::AbstractString...)
-    if after_exit[]
-        raw_print(msg...)
-    else
-        print_with_color(color, io, msg...)
```

Comment:
Hmm, the current `SimpleLogger` doesn't support these kind of tricks I'm afraid. Though perhaps it should if it's to be useful in debugging Base.

It's a bit of a tricky issue: it would be nice if general users should expect to be able to install their own `AbstractLogger`s, but it's not reasonable to expect them to have a fallback path with raw IO.

Maybe the trick is to have a fallback logger in `Base` (a simplified SimpleLogger, perhaps) which is somehow protected and reinstalled "after" exit, and falls back to ccall somewhat like you have here.

Reply:
Given how plain `print` is now broken in this case too (ie. you can't print from finalizers), I wouldn't worry about high-level logging not supporting that scenario either. Might look into it when I have another finalizer issue to debug 🙂